### PR TITLE
Feat: Renew Access Token

### PR DESCRIPTION
**Pull request by daffafaizan**
## Description

Added a function to renew the access token named ```renew_access_token```. As a refresher, this project uses JWT with two tokens, the ```refresh_token``` and the ```access_token``` that expires in **7 days** and **15 minutes** respectively.

Every **15 minutes** (recommended to renew before 15 minutes to avoid accidental logout in the front end), the ```access_token``` has to be renewed by:
1. Getting the ```refresh_token``` from the **cookie**
2. Validating the ```refresh_token``` and the **User** of the token.
3. Generating a new ``access_token```
4. Updating the **cookie** and the ``access_token``` field in the **User** object

The renewal process above can be done automatically by using the endpoint ```/v1/auth/renew```. The front end can use an interval system (every 10 minutes or so) to call this endpoint and renew the ```access_token```.

### CODE REVIEW

File: src/services/auth.py [hunk 1]
```diff
@@ -66,6 +74,42 @@ def login(response: Response, session: Session, payload: LoginSchema):
...
+        )
+        try:
+            user = get_user(session=session, field="refresh_token", value=refresh_token)
+
+            update_access_token(session, user, new_access_token)
+            response.set_cookie(
+                key="access_token", value=new_access_token, httponly=True
+            )
+            return new_access_token
+        except Exception:
```
Comment (jeremyalv) [inline src/services/auth.py:102]: If `get_user` throws a `CustomException`, we can specify that in addition to the generic `Exception` class.

**Review by jeremyalv — APPROVED** [review]

**Comment by jeremyalv** [comment]
Looks good. I think the process of re-authenticating the user is already clear in the code.

**Review by annavaws — APPROVED** [review]
Nice addition with the renew_access_token function to keep users logged in by automatically renewing their access tokens.

File: src/services/auth.py (hunk 1 above)
Comment (daffafaizan) [reply src/services/auth.py]: I made a custom exception class in the new ```exceptions``` module. Please let me know what you think.

File: src/controllers/auth.py
Comment (jeremyalv) [inline src/controllers/auth.py:1]: Can just remove `send_verif_email` and `validate_user_token` as it is included in the Email Verification pull request

**Pull request merged**